### PR TITLE
Bug: LT-1803 URL error message

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**

### CODE REVIEW

File: parties/serializers.py [hunk 1]
```diff
@@ -29,6 +30,16 @@ class Meta:
             "document",
         )
 
+    @staticmethod
+    def validate_website(value):
+        if value:
+            val = URLValidator()
+            url = f"https://{value}"
+            val(url)
```
Comment (markm5A8Z9F43913A) [inline parties/serializers.py]: If the URL is invalid, will this not throw an exception? Please add some tests for a `None`, a valid and an invalid URL.

File: parties/serializers.py [hunk 2]
```diff
@@ -29,6 +30,16 @@ class Meta:
...
+            return url
+        else:
+            return value
```
Comment (markm5A8Z9F43913A) [inline parties/serializers.py]: The `else` statement is probably not necessary the way the code is written now. But that's less important. Why do you want to return the invalid original value if it's empty?

File: parties/serializers.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply parties/serializers.py]: The string will not be verified if None/empty (the opening if statement) and it does throw exceptions for invalid URLs

File: parties/serializers.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply parties/serializers.py]: Because if it's empty we don't want to validate it (field is optional). The return could instead be `return ''` if you want

File: parties/serializers.py [hunk 3]
```diff
@@ -29,6 +30,16 @@ class Meta:
             "document",
         )
 
+    @staticmethod
+    def validate_website(value):
+        if value:
+            val = URLValidator()
```
Comment (stephanieboyd) [inline parties/serializers.py]: Call this `validator`, val is an abbreviation of value, and we've also a value argument

Comment (BenAndrew-Kainos) [reply parties/serializers.py]: done

File: parties/serializers.py [hunk 4]
```diff
@@ -29,6 +30,16 @@ class Meta:
             "document",
         )
 
+    @staticmethod
+    def validate_website(value):
+        if value:
+            validator = URLValidator()
+            url = f"https://{value}"
```
Comment (andytaylorkainos) [inline parties/serializers.py]: Might need to account for when the user actually did provide `https` (only inject `https` if it has not been provided)

File: parties/serializers.py [hunk 5]
```diff
@@ -29,6 +30,16 @@ class Meta:
             "document",
         )
 
+    @staticmethod
+    def validate_website(value):
+        if value:
```
Comment (stephanieboyd) [inline parties/serializers.py]: Gimme a docstring

File: parties/serializers.py (hunk 4 above)
Comment (BenAndrew-Kainos) [reply parties/serializers.py]: Just thought of that too :)

File: parties/serializers.py (hunk 5 above)
Comment (BenAndrew-Kainos) [reply parties/serializers.py]: done

File: parties/serializers.py (hunk 1 above)
Comment (markm5A8Z9F43913A) [reply parties/serializers.py]: Where is the exception being caught?

Comment (BenAndrew-Kainos) [reply parties/serializers.py]: It's doesn't need to be caught. You let the exception happen and that's how it sends the error back to the frontend

Comment (BenAndrew-Kainos) [reply parties/serializers.py]: So the behaviour with URlValidator() is the same and we're just making http/https optional

File: parties/serializers.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply parties/serializers.py]: Changed it to return '' for clarity
